src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/cache/FieldDataCacheRca.java

```diff
@@ -0,0 +1,204 @@
+ * from upstream metrics and maintains a collector which keeps track of the time window period(tp)
+ * where we repeatedly see evictions for the last tp duration. This RCA is marked as unhealthy if
+ * tp is above the threshold(300 seconds) and cache size exceeds the max cache size configured.
+ *
+ */
+public class FieldDataCacheRca extends Rca<ResourceFlowUnit<HotNodeSummary>> {
+    private static final Logger LOG = LogManager.getLogger(FieldDataCacheRca.class);
+    private static final long EVICTION_THRESHOLD_TIME_PERIOD_IN_MILLISECOND = TimeUnit.SECONDS.toMillis(300);
+
+    private final Metric fieldDataCacheEvictions;
+    private final Metric fieldDataCacheSize;
+    private final Metric fieldDataCacheMaxSize;
+    private final int rcaPeriod;
+    private int counter;
+    private boolean exceedsSize;
+    protected Clock clock;
+    private final CacheEvictionCollector cacheEvictionCollector;
+
+    public <M extends Metric> FieldDataCacheRca(final int rcaPeriod, final M fieldDataCacheEvictions,
+                                                final M fieldDataCacheSize, final M fieldDataCacheMaxSize) {
+        super(5);
+        this.rcaPeriod = rcaPeriod;
+        this.fieldDataCacheEvictions = fieldDataCacheEvictions;
+        this.fieldDataCacheSize = fieldDataCacheSize;
+        this.fieldDataCacheMaxSize = fieldDataCacheMaxSize;
+        this.counter = 0;
+        this.exceedsSize = Boolean.FALSE;
+        this.clock = Clock.systemUTC();
+        this.cacheEvictionCollector = new CacheEvictionCollector(FIELD_DATA_CACHE_EVICTION,
+                fieldDataCacheEvictions, EVICTION_THRESHOLD_TIME_PERIOD_IN_MILLISECOND);
+    }
+
+    @VisibleForTesting
+    public void setClock(Clock clock) {
+        this.clock = clock;
+    }
+
+    @Override
+    public ResourceFlowUnit<HotNodeSummary> operate() {
+        counter += 1;
+        long currTimestamp = clock.millis();
+
+        cacheEvictionCollector.collect(currTimestamp);
+        if (counter >= rcaPeriod) {
+            ResourceContext context;
+            HotNodeSummary nodeSummary;
+
+            ClusterDetailsEventProcessor.NodeDetails currentNode = ClusterDetailsEventProcessor.getCurrentNodeDetails();
+            double cacheSize = getTotalSizeInMB(fieldDataCacheSize);
+            double cacheMaxSize = getTotalSizeInMB(fieldDataCacheMaxSize);
+            exceedsSize = cacheMaxSize != 0 && cacheMaxSize != 0 && cacheSize > cacheMaxSize;
+            if (cacheEvictionCollector.isUnhealthy(currTimestamp) && exceedsSize) {
+                context = new ResourceContext(Resources.State.UNHEALTHY);
+                nodeSummary = new HotNodeSummary(currentNode.getId(), currentNode.getHostAddress());
+                nodeSummary.appendNestedSummary(cacheEvictionCollector.generateSummary(currTimestamp));
+            }
+            else {
+                context = new ResourceContext(Resources.State.HEALTHY);
+                nodeSummary = null;
```

Comment:
let's add node summary to healthy flowunit and send it as a heartbeat to cluster rca